__tests__/app.test.js

```diff
@@ -41,4 +42,26 @@ describe('app testing', () => {
             })
         })
     })
+    describe('get article by id', () => {
+        test('Returns the requested article', () => {
+            return request(app)
+            .get('/api/articles/2')
+            .expect(200)
+            .then(({ body: {article} }) => {
+                expect(article[0].article_id).toBe(2)
+                expect(article[0]).toEqual(
+                    expect.objectContaining({
+                        article_id: expect.any(Number),
+                        title: expect.any(String),
+                        topic: expect.any(String),
+                        author: expect.any(String),
+                        body: expect.any(String),
+                        created_at: expect.any(String),
+                        votes: expect.any(Number),
+                        article_img_url: expect.any(String)
+                    })
+                )
+            })
+        })
+    })
```

Comment:
There are also a couple of errors we need to test for and handle. What if the user requests an article but uses the an invalid ID (e.g. "ten" instead of "10")? Or what happens if the user uses a valid id (e.g. 99999) but that article doesn't exist in the database?

Reply:
I've added the error handling and managed to get them working, however i definitely feel like I've cheated. Needs reviewing.